src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Trigger.cs

```diff
@@ -60,17 +57,18 @@ public partial class ReindexDurableFunction
 
             // TODO: In .NET 5, use HttpRequestJsonExtensions.ReadFromJsonAsync which can gracefully handle different
             //       encodings. Here we'll expect a UTF-8 encoding which we can control as the client.
-            // We use Newtonsoft here for consistency across our other serializers
-            // (Although we could leverage the PipeReader if we used System.Text.Json)
             IReadOnlyList<int> extendedQueryTagKeys;
-            using (StreamReader streamReader = new StreamReader(request.Body, Encoding.UTF8, detectEncodingFromByteOrderMarks: true, bufferSize: 1024, leaveOpen: true))
-            using (JsonReader reader = new JsonTextReader(streamReader))
+            try
             {
-                extendedQueryTagKeys = DefaultJsonSerializer.Deserialize<IReadOnlyList<int>>(reader);
+                extendedQueryTagKeys = await JsonSerializer.DeserializeAsync<IReadOnlyList<int>>(request.Body, _jsonOptions, source.Token);
+                if (extendedQueryTagKeys == null || extendedQueryTagKeys.Count == 0)
+                {
+                    throw new JsonException("Expected at least 1 element in array.");
+                }
             }
-
-            if (extendedQueryTagKeys == null || extendedQueryTagKeys.Count == 0)
+            catch (JsonException e)
             {
+                logger.LogError(e, "Cannot deserialize extended query tag keys.");
                 return new HttpResponseMessage { StatusCode = HttpStatusCode.BadRequest };
             }
```

Comment:
https://media.giphy.com/media/mGK1g88HZRa2FlKGbz/giphy.gif

Reply:
lol I didn't want two different conditions to return `BadRequest`, so the deserialization may throw the exception... or I may do it directly 😛